[Field Formatters] Update color formatter UI

kowalczyk-krzysztof opened the pull request:
## Summary
This PR updates color formatter UI to match design suggested in https://github.com/elastic/kibana/issues/123527#issuecomment-1067087684

I have few questions @elastic/kibana-data-discovery :
1. I left prepend for inputs to be the same as the column names in previous implementations - should they be shortened? `Pattern (regular expression)` -> `Pattern`
`Range (min:max)` -> `Range`
2. The design from original comment mentions adding a help link - should this be added? If so, only for regex? Where would the link lead to?

Closes: #202620

kowalczyk-krzysztof committed:
Update color formatter UI

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

kertal commented:
> ## Summary
> This PR updates color formatter UI to match design suggested in [#123527 (comment)](https://github.com/elastic/kibana/issues/123527#issuecomment-1067087684)
> 
> I have few questions @elastic/kibana-data-discovery :

this is another nice change 👏 

> 
> 1. I left prepend for inputs to be the same as the column names in previous implementations - should they be shortened? `Pattern (regular expression)` -> `Pattern`
>    `Range (min:max)` -> `Range`

![CleanShot 2025-01-20 at 10 03 33](https://github.com/user-attachments/assets/ea52d781-e921-4b44-96f3-a09b38277ee6)

yes, I'd consider those as redundant

![CleanShot 2025-01-20 at 10 07 16](https://github.com/user-attachments/assets/5d1e3a7c-35d7-4257-bf07-08faea694590)

_To infinity and beyond!_
If we apply the same pattern like in the regexp having a default of min:max I think we could make it short. 

FYI @l-suarez , @ninoslavmiskovic 

> 2. The design from original comment mentions adding a help link - should this be added? If so, only for regex? Where would the link lead to?

good question, just found something for query DSL, an that's linking to wikipedia
https://www.elastic.co/guide/en/elasticsearch/reference/current/regexp-syntax.html
FYI @florent-leborgne, maybe you know, when in doubt we shouldn't add a link

kowalczyk-krzysztof committed:
Merge branch 'main' into feat/update-color-picker-ui

kowalczyk-krzysztof committed:
Shorten input prepend

kowalczyk-krzysztof commented:
> If we apply the same pattern like in the regexp having a default of min:max I think we could make it short.

I changed the input prepends to be shorter and I changed the default value of `range` to `min:max`.

> when in doubt we shouldn't add a link

I'll leave it as it is then for now, without a link.

florent-leborgne commented:
>good question, just found something for query DSL, an that's linking to wikipedia
https://www.elastic.co/guide/en/elasticsearch/reference/current/regexp-syntax.html
FYI @florent-leborgne, maybe you know, when in doubt we shouldn't add a link

This shouldn't require a link as long as it's made obvious that the field expects a regex (using the label or the placeholder text)

lukasolson reviewed (approved):
Tested & everything seems to work as desired.
Code changes LGTM.

davismcphee commented:
This seems worth backporting to 8.x. WDYT @kertal @lukasolson?
